src/jacobians.jl

```diff
@@ -1,14 +1,16 @@
-struct JacobianCache{CacheType1,CacheType2,CacheType3,fdtype,returntype,inplace}
+struct JacobianCache{CacheType1,CacheType2,CacheType3,ColorType,fdtype,returntype,inplace}
     x1  :: CacheType1
     fx  :: CacheType2
     fx1 :: CacheType3
+    color :: ColorType
 end
 
 function JacobianCache(
     x,
     fdtype     :: Type{T1} = Val{:central},
     returntype :: Type{T2} = eltype(x),
-    inplace    :: Type{Val{T3}} = Val{true}) where {T1,T2,T3}
+    inplace    :: Type{Val{T3}} = Val{true};
+    color = 1:length(x)) where {T1,T2,T3}
```

Comment:
```suggestion
    color = eachindex(x)) where {T1,T2,T3}
```

Reply:
I think we want to be able to assume an AbstractVector interface on it?

Reply:
so `eachindex` works in that case doesn't it? It's more generic than `1:length(x)` since it doesn't assume 1-index